### ISSUE TITLE
Image PMC4954863/JoU-2016-0019-g003.jpg not found in archive ftp://ftp.ncbi.nlm.nih.gov/pub/pmc/oa_package/2f/12/PMC4954863.tar.gz, skipping

### ISSUE THREAD

**Issue by 1112321sfdsaf** [razorx89/roco-dataset#9]
Thanks very much for sharing this wonderful dataset!
When downloading the dataset using the file 'fetch.py', I got this problem: 
![image](https://user-images.githubusercontent.com/75245666/192981637-356a4a56-4aae-4e35-a221-0552cded2749.png)
Can you give me some suggestions about it?

**Comment by saviola777** [comment]
There is no problem with the archive (I just tested it and for me there was no error). I am assuming there was a problem with the download. Try to re-run the script after it finishes (first making sure you delete any archives left in your download directory – `/tmp/roco-dataset/` by default). It should skip all images already extracted and hopefully extract the missing ones after re-downloading the problematic archives.

**Comment by 1112321sfdsaf** [comment]
Thanks for your quick reply, I will try again.

**Comment by 1112321sfdsaf** [comment]
I'm back.
I found some images existing in the "dlinks.txt" file are missing. So, I re-downloaded them using the provided links. 
However, I fail to download them.
For example, I used 'wget -r ftp://ftp.ncbi.nlm.nih.gov/pub/pmc/oa_package/32/34/PMC3999928.tar.gz' copying from 
![image](https://user-images.githubusercontent.com/75245666/193045166-f368f71f-ff12-4859-b6e8-0381d39eefdc.png)
and got some errors, the details are as follows.
![image](https://user-images.githubusercontent.com/75245666/193044957-6c05ff59-e2b5-412a-afce-b1d9a409b19a.png)

**Comment by saviola777** [comment]
It looks like some kind of connection issue between your machine and the PMC FTP server. For me the download works and the download speed is much higher as well. You should not need to download anything manually. Try to reduce the number of concurrent downloads and re-run the script until all images are there, maybe eventually it will work.

**Comment by 1112321sfdsaf** [comment]
Hi, I am back.

I ran "fetch.py" to download the whole dataset and repeated it a few times until no new images were updated.
Unfortunately, on the last download, there were still some images that failed to download.
![image](https://user-images.githubusercontent.com/75245666/194210135-92801030-03d6-47fe-87fa-ced415e88a93.png)
Moreover, I also counted the number of images. For the training set, I got 65,423 images; for the validation set, I got 8,174 images; and for the test set, I got 8,176 images. Are these images the whole dataset?

Thanks in advance for your reply.

**Comment by saviola777** [comment]
Some images that are not found are to be expected, since some papers receive updates over the years where the image names are changed. And yes, these ~81k images are the whole dataset.

**Comment by 1112321sfdsaf** [comment]
Thanks very much for your reply. I will close this issue.

**Comment by razorx89** [comment]
> Some images that are not found are to be expected, since some papers receive updates over the years where the image names are changed. And yes, these ~81k images are the whole dataset.

Are these images only renamed or are they missing? If they are only renamed, we could save a binary hash of the image file and search for it in case the image name is not found.

**Comment by saviola777** [comment]
I've looked at one example and the images were actually different, not just renamed – but it might still be possible to match them with perceptual image hashing using a low enough similarity threshold.
